content/en/blog/trust-chain/index.md

```diff
@@ -0,0 +1,362 @@
+
+Before I started to write the whole thing, I thought that I use our server-side
+secure enclave to store all the key pairs there and let the tenant set the
+enclave's master key. It would still mean that the implementation would be
+state-full. From the operations ' perspective, we all know what that means: more
+things to take care of and manage, but most importantly, one potential
+scalability issue to solve.
+
+The FIDO2 standard documentation describes a perfect solution for our needs
+which made our authenticator stateless. You give the other party your public
+key, but you give your private key in your `credential ID`. It might
+first sound crazy, but it's genius indeed.
+
+Hold on! That cannot be?
+
+But it is. You have to build your identifier to include your private key, but no
+one but you can use it because you have encrypted it with a symmetric master
+key. The key that no one but you controls.
+
+{{< imgproc MasterKeyMain.png Resize "1291x" >}}
+<em>Stateless Authenticator Implementation</em>
+{{< /imgproc >}}
+
+The draft above illustrates how our stateless FIDO2 authenticator works at
+a master key level. Other factors like a cloning counter and an authenticator ID
+are left out for simplicity. 
+1. We can ask TEE to create a new key pair for FIDO2 registration, which gives
+   us a unique key pair that includes `public key` and *encrypted private key*,
+   i.e.  `credential ID`.
+2. Our authenticator sends the FIDO2 `attestation object` to the server.
+3. When the authenticator receives the FIDO2 challenge during authentication, it builds
+   it to the key pair in the same format as registration.
+4. The TEE inside the authenticator builds us the `assertion object` ready to send
+   to the FIDO2 server.
+
+As we can see, the master key never leaves the TEE. The implementation can
+be done with help cloud
+[HSM](https://en.wikipedia.org/wiki/Hardware_security_module) or
+[TEE-based app](https://www.neclab.eu/research-areas/security/nec-labs-introduce-a-new-solution-enables-seamless-provisioning-and-decommissioning-of-tee-based-applications-in-the-cloud);
+or we can implement an application with the help of [AWS
+Nitro Enclaves](https://aws.amazon.com/ec2/nitro/nitro-enclaves/) or similar.
+
+**Note!** This is not a good solution for a pure client-side software-based
+authenticator. It's suitable for *hardware-based and certain types of
+server-side solutions* where you can use TEE or similar solutions.
+
+## Conclusion
+
+FIDO2 authentication is an excellent match for DID Wallet authentication. gRPC
+transport combined with JWT authorization has been straightforward to use. Our 
+gRPC SDK allows you to implicitly move the JWT token during the API calls after
+opening the server connection. Plus, gRPC's capability to have bidirectional
+streams make the programming experience very pleasant. Finally, an option is to
+authenticate the gRPC connection between server and client with self-signed TLS
```

Comment:
Remove "self-signed" or put it in parentheses.

Reply:
switched to "(no PKI is needed)"